tests/jballerina-unit-test/src/test/resources/test-src/error/error_test.bal

```diff
@@ -389,6 +389,16 @@ function testErrorBindingPattern() {
     assertEquality(b, false);
 }
 
+type ErrorDataWithErrorField record {
+    error 'error;
+};
+
+function testErrorDataWithErrorField() {
+    error newError = error("bam", message = "new error");
+    ErrorDataWithErrorField ef = {'error: newError};
+    assertEquality(ef.toString(), "{\"error\":error(\"bam\",message=\"new error\")}");
+}
+
```

Comment:
Shall we add a test case with error constructor with `error` as the arg name. Something like this `error e = error("msg", 'error = error("e"));`